src/fsharp/ConstraintSolver.fs

```diff
@@ -737,6 +737,10 @@ and SolveTypEqualsTyp (csenv:ConstraintSolverEnv) ndeep m2 (trace: OptionalTrace
     let sty1 = stripTyEqnsA csenv.g canShortcut ty1
     let sty2 = stripTyEqnsA csenv.g canShortcut ty2
 
+    match cxsln with
+    | Some (traitInfo, traitSln) when traitInfo.Solution.IsNone -> TransactMemberConstraintSolution traitInfo trace traitSln
```

Comment:
Yes, it is writing to a reference cell, see [this comments](https://github.com/gmpl/visualfsharp/blob/aec3cea9001aff7947770e88c31db48913e05714/src/fsharp/ConstraintSolver.fs#L1286) from the original code.